src/main/java/hudson/plugins/emailext/ExtendedEmailPublisher.java

```diff
@@ -352,7 +355,35 @@ private boolean sendMail(ExtendedEmailPublisherContext context) {
 
                     while (true) {
                         try {
-                            Transport.send(msg);
+                            ExtendedEmailPublisherDescriptor descriptor = getDescriptor();
+                            String smtpHost = descriptor.getSmtpServer();
+                            if (smtpHost == null) {
+                                smtpHost = "";
+                            }
+                            String awsRegex = "^email-smtp\\.([a-z0-9-]+)\\.amazonaws\\.com$";
+                            Pattern p = Pattern.compile(awsRegex);
+                            Matcher m = p.matcher(smtpHost);
+                            if (m.matches()) {
+                                String region = m.group(1);
+                                Session session = descriptor.createSession();
+                                Transport transport = session.getTransport(allRecipients[0]);
+                                try {
+                                    transport.connect();
+                                    transport.sendMessage(msg, allRecipients);
+                                    if (transport instanceof SMTPTransport) {
+                                        String response = ((SMTPTransport)transport).getLastServerResponse();
+                                        String[] parts = response.trim().split(" +");
+                                        if (parts.length == 3 && parts[0].equals("250") && parts[1].equals("Ok")) {
+                                            String MessageID = "<" + parts[2] + "@" + region + ".amazonses.com>";
+                                            msg.setHeader("Message-ID", MessageID);
+                                        }
```

Comment:
Ah, I was misreading the diff. I just really don't want to incorporate something specific like this. If it could be reworked into something generic, I would. You can always build a local version and use it as you are now, but unless its made more generic, I don't think I can include it.